internal/envoy/v3/route.go

```diff
@@ -121,6 +121,15 @@ func buildRoute(dagRoute *dag.Route, vhostName string, secure bool) *envoy_confi
 		// envoy.RouteRoute. Currently the DAG processor adds any HTTP->HTTPS
 		// redirect routes to *both* the insecure and secure vhosts.
 		route.Action = UpgradeHTTPS()
+
+		route.TypedPerFilterConfig = map[string]*anypb.Any{}
+		// Apply per-route authorization policy modifications.
+		if dagRoute.AuthDisabled {
+			route.TypedPerFilterConfig["envoy.filters.http.ext_authz"] = routeAuthzDisabled()
+		} else if len(dagRoute.AuthContext) > 0 {
+			route.TypedPerFilterConfig["envoy.filters.http.ext_authz"] = routeAuthzContext(dagRoute.AuthContext)
+		}
+
 	case dagRoute.DirectResponse != nil:
```

Comment:
Yes, I agree with you. we can even disable it totally. But right now it's configurable per route and one can enable/disable it as per their need.
But if you think that it will increase complexity of Contour's behavior I can disable it at all.

Reply:
I did not test this so I'm not 100% sure, but I'm assuming that currently two calls to authorization service will be generated for single client when client goes through the HTTPS upgrade? Once when hitting that route on the HTTP listener, and second time after redirect when the client accesses the actual route on HTTPS listener. If that happens, it sounds bit surprising to me. I would have thought there should be just single call, since the backing resource is accessed only once - and that was why I was thinking that should the HTTPS redirect be done without authorization. What do you think?

Reply:
I agree with you and I think we can disable these calls (before redirections).

I'll change the code.

Reply:
@tsaarni, It is fixed 🤝
You can review.